engine/src/main/java/org/hibernate/search/engine/common/impl/SearchIntegrationPartialBuildStateImpl.java

```diff
@@ -164,6 +169,17 @@ public SearchIntegration finalizeIntegration() {
 			}
 			failureCollector.checkNoFailure();
 
+			if ( previousIntegration.isPresent() ) {
```

Comment:
I'd call `preStart` all the time, and implement `preStart` in such a way that it behaves correctly in either case (reuse saved state if available, create missing resources otherwise). See my other comment about the `SavedState` abstraction.

The reason I would prefer that is to maximize the amount of code common to start and restart. Like I said the other time, the challenge here will be testing; the fewer alternate paths there are to test in the code, the better.

As to what goes in `preStart` and what goes into `start`: in my mind, everything that can be reused after a restart would be initialized in `preStart`, while the rest would be initialized in `start`. But I'm open to discussion about that, of course.